include/zephyr/net/net_if.h

```diff
@@ -944,6 +944,18 @@ struct net_if *net_if_get_default(void);
  */
 struct net_if *net_if_get_first_by_type(const struct net_l2 *l2);
 
+/**
+ * @brief Get nth network interface according to its type.
+ *
+ * @param l2 Layer 2 type of the network interface.
+ *
+ * @param node_pos position of the network interface in Layer 2 type list.
+ *
+ * @return nth network interface of a given type or NULL if no such
+ * interfaces was found.
+ */
+struct net_if *net_if_get_nth_by_type(const struct net_l2 *l2, int node_pos);
+
```

Comment:
I have some reservations for this function. The usecase for this seems to be very specific and depends on the interface position which can change if the interfaces are placed in the different order. Also all of this could be done in the application side which can use better heuristic to determine which interface to select.

Reply:
Removed it as per your suggestion